resources/sdk/pureclouddotnet/templates/ApiClient.mustache

```diff
@@ -156,6 +169,44 @@ namespace {{packageName}}.Client
             return request;
         }
 
+        private bool HasAccessTokenExpired(int statusCode)
+        {
+            TimeSpan timeSinceEpoch = (DateTime.UtcNow - new DateTime(1970, 1, 1));
+            long timeElapsed = (long) (timeSinceEpoch.TotalSeconds) - TokenIssueTime;
+            return statusCode == 401 && timeElapsed > TokenTimeToLive;
+        }
+
+        private void HandleExpiredAccessToken()
+        {
+            if (Monitor.TryEnter(RefreshTokenLock, 0))
```

Comment:
The object for locking may not be 100% correct here. Because the access token is stored on the `Configuration` instance, it's possible for two instances of `ApiClient` to use the same `Configuration` instance, which could result in multiple `ApiClient` instances attempting to refresh the same access token simultaneously; one will succeed and one will fail putting it in a bad state. If you use the Configuration object, it should prevent this.